### PR TITLE
Stronger check for DM password during server install

### PR DESCRIPTION
**Pull request by stlaz**
DM password is used as an NSS database password during server installation, therefore it must comply to NSS database requirements for password in FIPS.

https://fedorahosted.org/freeipa/ticket/5695

### CODE REVIEW

**Comment by pvoborni** [comment]
Function check_password_fips_nssdb_compatible  looks like a great candidate for unit test.

File: ipaserver/install/server/install.py [hunk 1]
```diff
@@ -54,6 +54,22 @@
...
 
 
+def check_password_fips_nssdb_compatible(password):
+    """
+    Check whether the given password can be used for NSSDB setup in FIPS mode
+    """
+    gotnumeric = any(c.isdigit() for c in password[:-1])
+    gotupper = any(c.isupper() for c in password[1:])
+    gotlower = any(c.islower() for c in password)
+    gotspecial = password.isalnum()
```
Comment (pvoborni) [inline ipaserver/install/server/install.py]: This line should be negated.

Comment (stlaz) [inline ipaserver/install/server/install.py]: Thanks, was editing this right before pushing but forgot how to logic :)

File: ipaserver/install/server/install.py [hunk 2]
```diff
@@ -54,6 +54,22 @@
 SYSRESTORE_DIR_PATH = paths.SYSRESTORE
 
 
+def check_password_fips_nssdb_compatible(password):
+    """
+    Check whether the given password can be used for NSSDB setup in FIPS mode
+    """
+    gotnumeric = any(c.isdigit() for c in password[:-1])
```
Comment (tomaskrizek) [inline ipaserver/install/server/install.py]: Perhaps we could use a comment that the `:-1` and `1:` is a NSSDB-specific requirement.

**Comment by HonzaCholasta** [comment]
IMHO you got it backwards - DM password may not comply to NSS requirements for passwords, therefore it must not be used as a password for any NSS database during server install.

**Comment by stlaz** [comment]
@HonzaCholasta: +1, you're right, I should investigate more on how to change this behavior, either we or Dogtag don't behave correctly here.
@pvoborni, @tomaskrizek: out of curiousity, do we have a design/guideline on how to write unit tests for FreeIPA? Did not find any.

**Comment by stlaz** [comment]
Closing as REJECTED, this will be sorted out in another way.

**Pull request closed without merging**